components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/tool-editor/views/tool-editor.js

```diff
@@ -82,9 +82,11 @@ define(['require', 'jquery', 'backbone', 'lodash', 'log', './design', "./source"
 
                     /* Start Debug Related Stuff */
                     var debugConfOpts = {
-                        debuggerInstance : self._sourceView.getDebugger(),
-                        editorInstance : self._sourceView.getEditor()
-                    }
+                        debugger_instance : self._sourceView.getDebugger(),
+                        editorInstance : self._sourceView.getEditor(),
+                        option : self.options.application.config.debugger_instance
```

Comment:
@AnuGayan why we need another option which gets the same debugger_instance ? isn't it the same as first attribute debugger_instance